features/messages/impl/src/main/kotlin/io/element/android/features/messages/impl/timeline/components/event/TimelineItemEncryptedView.kt

```diff
@@ -33,10 +35,17 @@ fun TimelineItemEncryptedView(
     onContentLayoutChanged: (ContentAvoidingLayoutData) -> Unit,
     modifier: Modifier = Modifier
 ) {
+    var text = stringResource(id = CommonStrings.common_waiting_for_decryption_key)
+    var iconId = CompoundDrawables.ic_compound_time
+    if (content.data is UnableToDecryptContent.Data.MegolmV1AesSha2 &&
+        content.data.utdCause == UtdCause.Membership) {
+        text = stringResource(id = CommonStrings.common_unable_to_decrypt_no_access)
+        iconId = CompoundDrawables.ic_compound_block
+    }
```

Comment:
It's a minor thing, but I think you could do something like so it's a bit clearer:

```kotlin
val isMembershipUtd = content.data is UnableToDecryptContent.Data.MegolmV1AesSha2 &&
        content.data.utdCause == UtdCause.Membership
val text = if (isMembershipUtd) {
    stringResource(id = CommonStrings.common_unable_to_decrypt_no_access)
} else {
    stringResource(id = CommonStrings.common_waiting_for_decryption_key)
}
val iconId = if (isMembershipUtd) {
    CompoundDrawables.ic_compound_block
} else {
    CompoundDrawables.ic_compound_time
}
```

Up to you though.

Reply:
Even more compact :) 
https://github.com/element-hq/element-x-android/pull/2754/commits/f16d2a34ef44ed505638baf11ae46f5eaad7075f